cv_next/app/cv/[cvId]/components/commentSection/comment.tsx

```diff
@@ -281,62 +298,32 @@ export default function Comment({
       commentToAdd.parent_comment_Id = comment.parent_comment_Id;
     }
 
-    const optimisticData = (
-      prevData: CommentModel[] | undefined
-    ): CommentModel[] => {
-      const newCommentsOfComments = new Map<string, CommentModel[]>(
-        prevData?.reduce((acc, cur) => {
-          // Use 'root' for top-level comments
-          const key = cur.parent_comment_Id || "root";
-          if (!acc.has(key)) {
-            acc.set(key, []);
-          }
-          acc.get(key)!.push(cur);
-          return acc;
-        }, new Map<string, CommentModel[]>()) || []
-      );
-
-      const parentId = comment.id;
-      if (!newCommentsOfComments.has(parentId)) {
-        newCommentsOfComments.set(parentId, []);
-      }
-      newCommentsOfComments
-        .get(parentId)!
-        .push(commentToAdd as unknown as CommentModel);
-
-      return Array.from(newCommentsOfComments.values()).flat();
-    };
+    mutate(comment.document_id, (currentSubComments: CommenterModel[] = []) => [
+      ...currentSubComments,
+      {
+        ...commentToAdd,
+        id: Date.now().toString(),
+        deleted: false,
+        last_update: new Date().toISOString(),
+        resolved: false,
+        upvoted: [],
+        user_id: userId,
+      },
+    ]);
 
     try {
+      await addComment(commentToAdd);
+      mutate(comment.document_id);
+    } catch (error) {
+      // Rollback optimistic update
       mutate(
         comment.document_id,
-        async () => {
-          await addComment(commentToAdd);
-          return optimisticData(commentsOfComment);
-        },
-        {
-          optimisticData: optimisticData(commentsOfComment),
-          rollbackOnError: (error) => {
-            if (error instanceof Error) {
-              return error.name !== "AbortError";
-            }
-            return true;
-          },
-          revalidate: true,
-        }
+        (currentComments: CommentModel[] = []) =>
+          currentComments.filter((comment) => comment.id !== comment.id),
+        false
```

Comment:
You don't really need to rollback if you don't use optimistic behavior

Reply:
See my reply above about this, if we decide that it isn't related I'll remove this